tests/test_Subfunction_Module.py

```diff
@@ -0,0 +1,320 @@
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    # Valid Cases
+    # Zero Array
+    Test_Signal_1 = np.zeros(10, dtype=int)
+    assert picant(Test_Signal_1, 0) == -1
+    # Sine Wave
+    Test_Signal_2 = np.sin(2 * np.pi * 5 * np.linspace(0, 2, 2 * 100))
+    assert picant(Test_Signal_2, 0) == -5
+
+    # Invalid Cases
+    with pytest.raises(ValueError):
+        picant(np.ones([2, 2]), 0)
+    with pytest.raises(ValueError):
+        picant(np.ones([2, 2, 2]), 0)
+    with pytest.raises(ValueError):
+        picant(np.array([]), 0)
+    with pytest.raises(ValueError):
+        picant(Test_Signal_1, -1)
+    return
+
+
+@pytest.mark.picpost_testmarker
+def test_picpost():
+    """
+    Test for the "picpost" function;
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    # Valid Cases
+    # Zero Array
+    Test_Signal_1 = np.zeros(10, dtype=int)
+    assert picpost(Test_Signal_1, 0) == 1
+    # Sine Wave
+    Test_Signal_2 = np.sin(2 * np.pi * 5 * np.linspace(0, 2, 2 * 100))
+    assert picpost(Test_Signal_2, 0) == 5
+
+    # Invalid Cases
+    with pytest.raises(ValueError):
+        picpost(np.ones([2, 2]), 0)
+    with pytest.raises(ValueError):
+        picpost(np.ones([2, 2, 2]), 0)
+    with pytest.raises(ValueError):
+        picpost(np.array([]), 0)
+    with pytest.raises(ValueError):
+        picpost(Test_Signal_1, -1)
+    return
+
+
+@pytest.mark.zerocros_testmarker
+def test_zerocros():
+    """
+    Test for the "zerocros" function;
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    # Valid Cases
+    Test_Signal_1 = np.zeros(10, dtype=int)
+    assert zerocros(Test_Signal_1) == 0
+    Test_Signal_2 = np.ones(10, dtype=int)
+    assert zerocros(Test_Signal_2).size == 0
+    Test_Signal_3 = np.ones(10, dtype=int) * (-1)
+    assert zerocros(Test_Signal_3).size == 0
+    Test_Signal_4 = np.linspace(9, -9, 19)
+    assert zerocros(Test_Signal_4) == 9
+
+    # Invalid Cases
+    with pytest.raises(ValueError):
+        zerocros(np.ones([2, 2]))
+    with pytest.raises(ValueError):
+        zerocros(np.ones([2, 2, 2]))
+    with pytest.raises(ValueError):
+        zerocros(np.array([]))
+    return
+
+
+@pytest.mark.modmax_testmarker
+def test_modmax():
+    """
+    Test for the "modmax" function;
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    # Valid Cases
+    # Zero Array
+    Test_Signal_1 = np.zeros(10, dtype=int)
+    assert modmax(Test_Signal_1, 0, 0, 1).size == 0
+    assert modmax(Test_Signal_1, 0, 0, -1).size == 0
+    assert modmax(Test_Signal_1, 0, 0, 0).size == 0
+
+    # Sine Wave
+    Test_Signal_2 = np.sin(2 * np.pi * 5 * np.linspace(0, 2, 2 * 100))
+    assert modmax(Test_Signal_2, 0, 2, 1).size == 0
+    assert modmax(Test_Signal_2, 0, 2, -1).size == 0
+    assert modmax(Test_Signal_2, 0, 2, 0).size == 0
+    assert modmax(Test_Signal_2, len(Test_Signal_2), 2, 0).size == 0
+    assert np.array_equal(
+        modmax(Test_Signal_2, 0, 0, 1),
```

Comment:
You can `squeeze()` the output of the `modmax` and then you won't need to deal with the dimensions of the reference array.